packages/docusaurus-plugin-content-blog/src/__tests__/blogUtils.test.ts

```diff
@@ -301,19 +301,9 @@ describe('linkify', () => {
 
 const createBlogPost = (args: DeepPartial<BlogPost>): BlogPost => ({
   id: '',
-  metadata: {
-    source: '',
-    title: '',
-    formattedDate: '',
-    permalink: '',
-    description: '',
-    hasTruncateMarker: false,
-    authors: [],
-    frontMatter: {},
-    tags: [],
-    unlisted: false,
+  metadata: fromPartial({
```

Comment:
fromPartial supports deep partials afaik so you should be able to apply it at the very top

```js
const createBlogPost = (blogPost: DeepPartial<BlogPost>): BlogPost => fromPartial(blogPost)
```

Depending on calling code, we might even not need this function anymore, you can just do:

```ts
const blogPost: BlogPost = fromPartial({})
```

or

```ts
fromPartial<BlogPost>({})
```